clang/lib/Driver/Driver.cpp

```diff
@@ -5913,11 +5888,24 @@ class OffloadingActionBuilder final {
 
           // Select remangled libclc variant
           std::string LibSpirvTargetName =
-              (TC->getAuxTriple()->isOSWindows())
-                  ? "remangled-l32-signed_char.libspirv-nvptx64-nvidia-cuda."
-                    "bc"
-                  : "remangled-l64-signed_char.libspirv-nvptx64-nvidia-cuda."
-                    "bc";
+
+              isNativeCPU ?
+                          // Select libclc variant based on target triple.
+                          // On Windows long is 32 bits, so we have to select
+                          // the right remangled libclc version.
+                  ((TC->getAuxTriple()->isOSWindows()
+                        ? "remangled-l32-signed_char.libspirv-"
+                        : "remangled-l64-signed_char.libspirv-") +
+                   TC->getTripleString() + ".bc")
```

Comment:
I believe the target triple for NVPTX is expected to be `nvptx64-nvidia-cuda`, so we should be able to use the `TC->getTripleString()` for all cases.  If that's not the case, maybe use a variable to hold the triple string to use instead of this cascading value assignment for readibility.

Reply:
Using `TC->getTripleString()` for all cases would change the current behavior because `Triple::isNVPTX()` also returns true for target triple `nvptx-nvidia-cuda`. I'm not sure if this triple is meant to be supported but it's not rejected. Both, `nvptx` and `nvptx64` currently select the same libclc (for nvptx64), which may be a separate issue. But to keep this behavior I've created a variable to hold the target triple as you suggested. Is that change acceptable?